src/utils/AppRouteProvider.jsx

```diff
@@ -37,6 +48,19 @@ const AppRouteProvider = (props) => {
       setRenderIntercept(availableIntercepts.DATA_PROCESSING(nextRoute, hardNavigate));
       return;
     }
+
+    if (previousRoute.match('/data-management')) {
+      const lastViewed = moment().toISOString();
```

Comment:
I would prefer you put all the new code inside a new function (with some comments of why it's needed) in this file so the `AppRouteProvider` remains easy to read. e.g.:
```
if (previousRoute.match('/data-management')) {
     updateExperiments();
     continueNavigation(nextRoute, hardNavigate);
}
```

Reply:
refactored